src/gpu_sysman.c

```diff
@@ -768,16 +771,44 @@ static int gpu_init(void) {
 }
 
 /* Add device labels to all metrics in given metric family and submit family to
- * collectd.  Resets metric family after dispatch */
+ * collectd, and log the metric if metric logging is enabled.
+ * Resets metric family after dispatch */
 static void gpu_submit(gpu_device_t *gpu, metric_family_t *fam) {
+  const char *pci_bdf = gpu->pci_bdf;
+  struct timespec ts = {0, 0};
+  if (config.logmetrics) {
+    clock_gettime(CLOCK_MONOTONIC, &ts);
+    /* skip common BDF address prefix */
+    if (strncmp("0000:", pci_bdf, 5) == 0) {
+      pci_bdf += 5;
+    }
+  }
```

Comment:
It looks like the `ts` and `pci_bdf` variables are only used if `config.logmetrics` is true. I think it would be more readable if you initialized them unconditionally.

Reply:
Done.